lib/preprocessor.js

```diff
@@ -125,7 +120,12 @@ function createPreprocessor (config, basePath, injector) {
           }
 
           instances[name] = p
-          preprocessors.push(p)
+          if (!thisFileIsBinary || p.handleBinaryFiles) {
```

Comment:
Do we really need `handleBinaryFiles` flag and this whole if statement ? Maybe we can just allow to preprocess binary files, and if something fails, preprocessor will throw error.

I guess in more than 90% there will be no binary files that matches pattern and should be preprocessed (I may miss something).

Just thinking out loud, and I'm curious others opinion about that.

Reply:
This is just for safe backwards compatibility.
Currently if there is a pattern `foo/*` that matches binary files, it is ignored. Without this flag, the 10% projects that do include binary files in their pattern will break.